packages/back-end/src/integrations/ClickHouse.ts

```diff
@@ -28,28 +28,31 @@ export default class ClickHouse extends SqlIntegration {
     return ["password"];
   }
   async runQuery(sql: string): Promise<QueryResponse> {
-    const client = new ClickHouseClient({
-      url: this.params.url,
-      port: this.params.port,
-      basicAuth: this.params.username
-        ? {
-            username: this.params.username,
-            password: this.params.password,
-          }
-        : null,
-      format: "json",
-      debug: false,
-      raw: false,
-      config: {
-        database: this.params.database,
-      },
-      reqParams: {
-        headers: {
-          "x-clickhouse-format": "JSON",
-        },
+    const client = createClient({
+      host: `${this.params.url}${
+        this.params.port ? `:${this.params.port}` : ""
+      }`,
```

Comment:
The old Clickhouse client supported specifying a port in the URL while also passing in a `port` setting.  For example, this would work fine before, but would break today:

```ts
{
  url: "http://localhost:1234",
  port: "1234"
}
```

There was also some weird behavior in the old client if you passed in different ports in both places, but I don't think we need to support that edge case.

Reply:
An easy solution to this is to use the URL class.

```ts
function getHost(url: string | undefined, port: number) {
  if (!url) return undefined;
  const host = new URL(url);
  if (!host.port && port) host.port = port + "";
  return host.origin;
}
// ...

const client = createClient({
  host: getHost(this.params.url, this.params.port),
```

Reply:
One issue with this is that `URL(url)` will throw an error if `url` isn't a proper URL. Should I just catch this case and do my pasting approach or should we consider some better validation when the form is saved (I don't think we do any right now).

Reply:
for example, do something like this to catch some edge cases

```
export function getHost(url: string | undefined, port: number): string | undefined {  
  if (!url) return undefined;  
  try {
    const host = new URL(url);  
    if (!host.port && port) host.port = port + "";  
    return host.origin;  
  } catch (e) {
    return `${url}:${port}`;
  }
}  
```

Reply:
I think it's fine to just let it throw.  I'm pretty sure if they have an invalid URL today it will be broken, so this shouldn't introduce any regressions.